src/data/polynomial/basic.lean

```diff
@@ -55,6 +57,9 @@ The embedding from `R` is called `C`. -/
 structure polynomial (R : Type*) [semiring R] := of_finsupp ::
 (to_finsupp : add_monoid_algebra R ℕ)
 
+-- do not unfold polynomial in simps lemmas
+initialize_simps_projections polynomial (-to_finsupp)
```

Comment:
This doesn't seem to help. With my change, `@[simps]` on `poly_equiv_tensor.to_fun_biliear` generates `to_fun_bilinear_apply_apply_to_finsupp_to_fun` instead of the `to_fun_bilinear_apply_apply` it did before.